distributions/aws/test/e2e/utils/cloudformation_resources.py

```diff
@@ -0,0 +1,131 @@
+from e2e.utils.config import configure_resource_fixture
+
+
+def create_stack(
+    cfn_client,
+    template_path,
+    stack_name,
+    params={},
+    capabilities=[],
+    timeout=300,
+    interval=10,
+):
+    """
+    Creates a cloudformation stack from a cloudformation template
+    """
+
+    with open(template_path) as file:
+        body = file.read()
+
+    stack_params = [
+        {"ParameterKey": key, "ParameterValue": value, "UsePreviousValue": False}
+        for key, value in params.items()
+    ]
+
+    resp = cfn_client.create_stack(
+        StackName=stack_name,
+        TemplateBody=body,
+        Parameters=stack_params,
+        Capabilities=capabilities,
+    )
+
+    stack_id = resp["StackId"]
+
+    print(f"Creating stack {stack_name}, waiting for stack create complete")
+    waiter = cfn_client.get_waiter("stack_create_complete")
+    waiter.wait(
+        StackName=stack_name,
+        WaiterConfig={"Delay": interval, "MaxAttempts": timeout // interval + 1},
+    )
+
+    return stack_name, stack_id
+
+
+def delete_stack(cfn_client, stack_name, timeout=300, interval=10):
+    """
+    Deletes a cloudformation stack
+    """
+
+    cfn_client.delete_stack(StackName=stack_name)
+
+    print(f"Deleting stack {stack_name}, waiting for stack delete complete")
+    waiter = cfn_client.get_waiter("stack_delete_complete")
+    waiter.wait(
+        StackName=stack_name,
+        WaiterConfig={"Delay": interval, "MaxAttempts": timeout // interval + 1},
+    )
+
+
+def describe_stack(cfn_client, stack_name):
```

Comment:
if you make cfn_client a method then no longer need to pass them

Reply:
yeah was just following a functional programming style, i can include a class if an OO approach